Assets/Plugins/LeapMotion/Modules/Hands/Scripts/HandBinder/Editor/HandBinderEditor.cs

```diff
@@ -0,0 +1,718 @@
+            //Setting a new bone has to be done when the hand is not an in editor pose, so doing this will reset the hand
+            if(setEditorPose.boolValue == true) {
+                myTarget.ResetHand();
+                setEditorPose.boolValue = false;
+            }
+
+            var startTransform = boneProperty.FindPropertyRelative("startTransform");
+
+            startTransform.FindPropertyRelative("position").vector3Value = boundTransform.localPosition;
+            startTransform.FindPropertyRelative("rotation").vector3Value = boundTransform.localRotation.eulerAngles;
+        }
+
+        /// <summary>
+        /// Set the Editor Pose
+        /// </summary>
+        private void EditorHandPose() {
+            if(myTarget != null && myTarget.enabled) {
+                MakeLeapHand(myTarget);
+
+                if(myTarget.SetEditorPose) {
+                    if(myTarget.GetLeapHand() == null) {
+                        myTarget.InitHand();
+                        myTarget.BeginHand();
+                        myTarget.UpdateHand();
+                    }
+                    else {
+                        myTarget.UpdateHand();
+                    }
+                }
+                else {
+                    myTarget.ResetHand();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Makes a new leap hand so we can use it to set an editor pose
+        /// </summary>
+        /// <param name="binder"></param>
+        private void MakeLeapHand(HandBinder binder) {
+            LeapProvider provider = null;
+
+            //First try to get the provider from a parent HandModelManager
+            if(binder.transform.parent != null) {
+                var manager = binder.transform.parent.GetComponent<HandModelManager>();
+                if(manager != null) {
+                    provider = manager.leapProvider;
+                }
+            }
+
+            //If not found, use any old provider from the Hands.Provider getter
+            if(provider == null) {
+                provider = Hands.Provider;
+            }
+
+            Hand hand = null;
+            //If we found a provider, pull the hand from that
+            if(provider != null) {
+                var frame = provider.CurrentFrame;
+
+                if(frame != null) {
+                    hand = frame.Get(binder.Handedness);
+                }
+            }
+
+            //If we still have a null hand, construct one manually
+            if(hand == null) {
+                hand = TestHandFactory.MakeTestHand(binder.Handedness == Chirality.Left, unitType: TestHandFactory.UnitType.LeapUnits);
+                hand.Transform(binder.transform.GetLeapMatrix());
+            }
+
+            binder.LeapHand = hand;
+        }
+
+        /// <summary>
+        /// Converts
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
```

Comment:
Could make create a dictionary of BoundType to a tuple:

```
Dictionary<BoundType, (Finger.FingerType, Finger.BoneType)> map = new Dictionary<BoundType, (Finger.FingerType, Finger.BoneType)>{
    {BoundTypes.THUMB_METACARPAL, (Finger.FingerType.TYPE_THUMB, Bone.BoneType.TYPE_METACARPAL)}
    //etc
};

private SerializedProperty BoundTypeToProperty(int id)
{
    SerializedProperty offsetProperty = null;
    var boundType = (BoundTypes)id;
   if(!map.ContainsKey(boundType))
        return null;

    var (fingerType, boneType) = map[boundType];
    return OffsetPropertyFromLeapTypes(fingerType, boneType);
}
```

Reply:
Fantastic idea, this cleaned up the code a lot and I was looking for a solution like this.